File: Level 1 / Python /banking_system.py
# banking_system.py

from account import Account
from transactions import DepositTransaction, TransferTransaction

class BankingSystem:
    def __init__(self):
        self.accounts = {}

    def create_account(self, timestamp: int, account_id: str) -> bool:
        if account_id in self.accounts:
            return False
        self.accounts[account_id] = Account(account_id)
        return True

    def deposit(self, timestamp: int, account_id: str, amount: int) -> int | None:
        transaction = DepositTransaction(timestamp, account_id, amount)
        return transaction.execute(self.accounts)

    def transfer(self, timestamp: int, source_id: str, target_id: str, amount: int) -> int | None:
        transaction = TransferTransaction(timestamp, source_id, target_id, amount)
        return transaction.execute(self.accounts)

File: Level 1 /C#/bankingSystem.cs
using System.Collections.Generic;

public class BankingSystem
{
    private readonly Dictionary<string, Account> _accounts = new();

    public bool CreateAccount(int timestamp, string accountId)
    {
        if (_accounts.ContainsKey(accountId))
            return false;

        _accounts[accountId] = new Account(accountId);
        return true;
    }

    public int? Deposit(int timestamp, string accountId, int amount)
    {
        var transaction = new DepositTransaction(timestamp, accountId, amount);
        return transaction.Execute(_accounts);
    }

    public int? Transfer(int timestamp, string sourceId, string targetId, int amount)
    {
        var transaction = new TransferTransaction(timestamp, sourceId, targetId, amount);
        return transaction.Execute(_accounts);
    }
}
